signer/tests/integration/transaction_coordinator.rs

```diff
@@ -4712,3 +4714,89 @@ mod get_eligible_pending_withdrawal_requests {
         testing::storage::drop_db(db).await;
     }
 }
+
+#[test_log::test(tokio::test)]
+async fn should_handle_dkg_coordination_failure() {
+    let mut rng = get_rng();
+    let mut context = TestContext::builder()
+        .with_in_memory_storage()
+        .with_mocked_clients()
+        .build();
+
+    let storage = context.get_storage_mut();
+
+    // Create a bitcoin block to serve as chain tip
+    let bitcoin_block: model::BitcoinBlock = Faker.fake_with_rng(&mut rng);
+    storage.write_bitcoin_block(&bitcoin_block).await.unwrap();
+
+    // Get chain tip reference
+    let chain_tip = storage
+        .get_bitcoin_canonical_chain_tip_ref()
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Create a set of signer public keys and update the context state
+    let mut signer_keys = BTreeSet::new();
+    for _ in 0..3 {
+        // Create 3 signers
+        let private_key = PrivateKey::new(&mut rng);
+        let public_key = PublicKey::from_private_key(&private_key);
+        signer_keys.insert(public_key);
+    }
+    context
+        .state()
+        .update_current_signer_set(signer_keys.clone());
+
+    // Mock the stacks client to handle contract source checks
+    context
+        .with_stacks_client(|client| {
+            client.expect_get_contract_source().returning(|_, _| {
+                Box::pin(async {
+                    Ok(ContractSrcResponse {
+                        source: String::new(),
+                        publish_height: 1,
+                        marf_proof: None,
+                    })
+                })
+            });
+        })
+        .await;
+
+    // Verify DKG should run
+    assert!(
+        transaction_coordinator::should_coordinate_dkg(&context, &chain_tip)
+            .await
+            .unwrap(),
+        "DKG should be triggered since no shares exist yet"
+    );
+
+    // Create coordinator with test parameters using SignerNetwork::single
+    let network = SignerNetwork::single(&context);
+    let mut coordinator = TxCoordinatorEventLoop {
+        context: context.clone(),
+        network: network.spawn(),
+        private_key: PrivateKey::new(&mut rng),
+        threshold: 3,
+        context_window: 5,
+        signing_round_max_duration: std::time::Duration::from_secs(5),
+        bitcoin_presign_request_max_duration: std::time::Duration::from_secs(5),
+        // extremely short duration so that DKG fails...
+        dkg_max_duration: Duration::from_millis(1),
```

Comment:
I would probably set this to like 1ms or something to make absolutely certain that no combination of config/fast computer can cause this to succeed.